pubpub/platform
# Add pub and stage membership management

## Issue(s) Resolved
Resolves #594 
Resolves #595 

## High-level Explanation of PR
- Refactors the add member dialog to work with pub and stage memberships
- Refactors the add member dialog to no longer be controlled through url path and query parameters
	- 	Instead of fetching suggested users in a server component and rerendering, it now uses [React Query](https://tanstack.com/query/v5/docs/framework/react/quick-start) through the [ts-rest generated client](https://ts-rest.com/docs/react-query/v5-setup) to request suggestions via fetch + an api endpoint
	- This should make it more portable and more performant
	- It's no longer possible to link directly to the member add dialog, and it won't keep state after a refresh, which I think are unnecessary features. LMK if not and I can bring them back for the members page only.
- Adds the ability to change the role and remove members from the existing pub and stage member lists
- Modifies the user search endpoint to only return a user when the email matches exactly. This was always the intention from a privacy perspective, but also fixed a bug which could in some cases prevent inviting a new user if their email partially matched an existing one.

## Test Plan
- As jill admin:
   - Visit the members page and ensure you can still invite and remove members as usual
   - Visit the stage editor, open a stage, and select the stages panel.
   - Click add member and try adding a member that exists in the community
   - Click add member and try adding that member again. You shouldn't be able to
   - Click add member and try adding a member that doesn't exist in the community. Check inbucket to make sure the invite email looks ok. The signup process should work, although they won't be a member of a community so don't expect to be able to navigate beyond the settings page after logging in.
   - Make sure you can delete a member and change their role
   - Visit an individual pub and run the same tests!
- as some@pubpub.org:
    -  Visit the same individual pub that you added some users to above. You should see the members, but not be able to edit them
   
## Screenshots

Stage member management
<img width="1624" alt="image" src="https://github.com/user-attachments/assets/42707326-0657-4249-a500-23a4d457638e">

Pub member management
<img width="1624" alt="image" src="https://github.com/user-attachments/assets/be25b306-71f0-4ef2-b092-966103ab2450">

Pub members in readonly mode
<img width="1624" alt="image" src="https://github.com/user-attachments/assets/d65e92c4-c60e-4aa4-8449-f3b87ca37d3e">

Side by side comparison of width change on pub page:
![image](https://github.com/user-attachments/assets/7208bbe4-e475-4b70-b3fd-ba023f97d943)
